gotoalert/events.py

```diff
@@ -431,3 +445,99 @@ def get_skymap(self, nside=128):
             self.contour_areas[contour] = self.skymap.get_contour_area(contour)
 
         return self.skymap
+
+class NUEvent(Event):
+    """A class to represent a Gamma-Ray Burst Event."""
+
+    def __init__(self, payload):
+        super().__init__(payload)
+
+        # Get XML param dicts
+        # NB: you can't store these on the Event because they're unpickleable.
+        top_params = vp.get_toplevel_params(self.voevent)
+        group_params = vp.get_grouped_params(self.voevent)
+
+        # Default params
+        self.notice = EVENT_DICTONARY[self.packet_type]['notice_type']
+        self.type = EVENT_DICTONARY[self.packet_type]['event_type']
+        self.source = EVENT_DICTONARY[self.packet_type]['source']
+
+        # Get the run and event ID (e.g. 13311922683750)
+        self.id = top_params['AMON_ID']['value']
+
+        # Create our own event name (e.g. ICECUBE_GOLD_EVENT_13311922683750)
+        self.name = '{}_{}_{}'.format(self.source, self.notice, self.id)
+
+        # Get info from the VOEvent
+        # signalness: the probability this is an astrophysical signal relative to backgrounds
+        self.signalness = float(top_params['signalness']['value'])
+        self.far = float(top_params['FAR']['value'])
+
+        # Position coordinates
+        self.position = vp.get_event_position(self.voevent)
+        self.coord = SkyCoord(ra=self.position.ra, dec=self.position.dec, unit=self.position.units)
+        self.target = FixedTarget(self.coord)
+
+        # Position error
+        self.coord_error = Angle(self.position.err, unit=self.position.units)
+        
+        # Systematic error for cascade event is given, so = 0
+        if self.notice == 'ICECUBE_CASCADE':
+            self.systematic_error = Angle(0, unit='deg')
+        else:
+            self.systematic_error = Angle(.2, unit='deg')
+        self.total_error = Angle(np.sqrt(self.coord_error ** 2 + self.systematic_error ** 2),
+                                 unit='deg')
+
+        # Enclosed skymap url for CASCADE_EVENT, but others
+        # Get skymap URL
+        for group in group_params:
+            if 'skymap_fits' in group_params[group]:
+                self.skymap_url = group_params[group]['skymap_fits']['value']
+                self.skymap_type = group
+            else:
+                self.skymap_url = None
```

Comment:
You're not actually reading the skymap here, because the 'skymap_fits' param is in `top_params`, not in a group.

```suggestion
        if 'skymap_fits' in top_params:
            self.skymap_url = top_params['skymap_fits']['value']
        else:
            self.skymap_url = None
```